sql: add database name to crdb_internal.tables

madelynnblue opened the pull request:
Without this it's not possible to select the correct row from
crdb_internal.tables since there's no non-root way to find out the
database ID of a table by name.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13836)
<!-- Reviewable:end -->

knz commented:
LGTM with a small code suggestion.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/crdb_internal.go, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13836#-Ke4Cb6N0K5uER3puni_:-Ke4Cb6N0K5uER3punia:bd0e3t5) ([raw file](https://github.com/cockroachdb/cockroach/blob/e038dea4fb2806a298fee4400d3b02463938ca3c/pkg/sql/crdb_internal.go#L58)):*
> ```Go
> 			return err
> 		}
> 		dbDescs := make(map[sqlbase.ID]*sqlbase.DatabaseDescriptor)
> ```

Since you only need the name I would suggest making the code more readable/simple by using a `map[sqlbase.ID]string` and store only the names in the map.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13836#-:-Ke4CnKJiIEbDVp667o6:bom8pky)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: add database name to crdb_internal.tables

Without this it's not possible to select the correct row from
crdb_internal.tables since there's no non-root way to find out the
database ID of a table by name.

mjibson commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/crdb_internal.go, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13836#-Ke4Cb6N0K5uER3puni_:-Ke4kcbX-RkQc7-wkqxn:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e038dea4fb2806a298fee4400d3b02463938ca3c/pkg/sql/crdb_internal.go#L58)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Since you only need the name I would suggest making the code more readable/simple by using a `map[sqlbase.ID]string` and store only the names in the map.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13836)*
<!-- Sent from Reviewable.io -->
